### PR TITLE
Implement importing a list of files

### LINKED ISSUE #890: Allow Import to accept a delimited list of Project files
I'd like to build up a list of project files to import in an ItemList and import them. So, for example, this:

```
<Import Project="Foo.props" />
<Import Project="Bar.props" />
```

 would be equivalent to this:

```
<ItemGroup>
  <Projects Include="Foo.props;Bar.props"/>
</ItemGroup>
<Import Project="@(Projects)" />
```

Seems a natural construct... 

Warm regards,
Chris

### PR DESCRIPTION
**Pull request by jeffkl**
Allows `<Import Project="import1.props;import2.props" />` to work.  This also updates `Exists` so that this will work:

```xml
<Import Project="import1.props;import2.props"
   Condition="Exists('import1.props;import2.props')" />
```

It also works with properties:

```xml
<Project>

  <PropertyGroup>
    <Imports>import1.props</Imports>
    <Imports>$(Imports);import2.props</Imports>
  </PropertyGroup>

  <Import Project="$(Imports)" Condition=" '$(Imports)' != '' And Exists($(Imports))" />
</Project>
```
**It does not work with items** because items are not available when evaluating `<Import />` elements.

Closes #890

### CODE REVIEW

File: src/Build/Evaluation/Conditionals/FunctionCallExpressionNode.cs [hunk 1]
```diff
@@ -38,20 +38,11 @@ internal override bool BoolEvaluate(ConditionEvaluator.IConditionEvaluationState
...
-
                 try
                 {
-                    if (state.EvaluationDirectory != null && !Path.IsPathRooted(expandedValue))
-                    {
-                        expandedValue = Path.GetFullPath(Path.Combine(state.EvaluationDirectory, expandedValue));
-                    }
+                    return !ExpandArgumentAsFileList((GenericExpressionNode) _arguments[0], state)
+                        .DefaultIfEmpty()
+                        .Any(i => i == null || state.LoadedProjectsCache?.TryGet(i) == null && !FileUtilities.FileOrDirectoryExistsNoThrow(i));
```
Comment (cdmihai) [inline src/Build/Evaluation/Conditionals/FunctionCallExpressionNode.cs]: However much it hurts me to say this, I'd advise against Linq and variable capturing lambdas inside evaluation code. :(
This is not the hottest of paths, but it's still evaluation, which runs a lot. RPS might hiccup here.

Comment (cdmihai) [inline src/Build/Evaluation/Conditionals/FunctionCallExpressionNode.cs]: Should it be Any or All? I'd keep Exists to check for all, and maybe add an ExistsAny function to check for any.

Comment (cdmihai) [inline src/Build/Evaluation/Conditionals/FunctionCallExpressionNode.cs]: In `Exists("foo.cs;bar.cs")` I'd expect Exists to check that both files exist.

---
In reply to: [113551386](https://github.com/Microsoft/msbuild/pull/2014#discussion_r113551386) [](ancestors = 113551386)

Comment (cdmihai) [inline src/Build/Evaluation/Conditionals/FunctionCallExpressionNode.cs]: Oops, the conditions inside the Any are actually negated, didn't see that. Could add a comment warning readers :)

---
In reply to: [113551764](https://github.com/Microsoft/msbuild/pull/2014#discussion_r113551764) [](ancestors = 113551764,113551386)

Comment (cdmihai) [inline src/Build/Evaluation/Conditionals/FunctionCallExpressionNode.cs]: From a discussion with Jeff, this is actually not that bad. The lambda object is created once per method call, and the Linq expression isn't that long.

---
In reply to: [113551198](https://github.com/Microsoft/msbuild/pull/2014#discussion_r113551198) [](ancestors = 113551198)

File: src/Build.UnitTests/ExpressionTreeExpression_Tests.cs [hunk 2]
```diff
@@ -336,11 +336,6 @@ public class ExpressionTest : IDisposable
             "existsX",
             "!",
             "nonexistentfunction('xyz')",
-            "exists('a;b')", /* non scalar */
```
Comment (cdmihai) [inline src/Build.UnitTests/ExpressionTreeExpression_Tests.cs]: Make sure to add a test where the semicolon is escaped

Comment (jeffkl) [inline src/Build.UnitTests/ExpressionTreeExpression_Tests.cs]: This test already does that.

**Comment by AndyGerlicher** [comment]
Looks good! We'll hold until Monday or Tuesday, this shouldn't go in the first preview of Update 3 and we haven't branched yet.

**Pull request merged**